test/runtests.jl

```diff
@@ -809,3 +810,20 @@ Base.similar(bc::Broadcast.Broadcasted{Broadcast.ArrayStyle{MyArray}}, ::Type{El
     s = StructArray{ComplexF64}((MyArray(rand(2,2)), MyArray(rand(2,2))))
     @test_throws MethodError s .+ s
 end
+
+@testset "staticarrays" begin
+    # test broadcast + components for vectors
+    for StaticVectorType = [SVector, MVector, SizedVector]
+        x = StructArray([StaticVectorType{2}(Float64[i;i+1]) for i = 1:2])
```

Comment:
Could you also test that this infers correctly? I think simply

```julia
@inferred StructArray(somearrayofstaticvectors)
```

should do it.

Reply:
Sounds good, added.